awx/main/management/commands/inventory_import.py

```diff
@@ -75,7 +76,27 @@ def get_base_args(self):
         bargs.extend(['-v', '{0}:{0}:Z'.format(self.source)])
         for key, value in STANDARD_INVENTORY_UPDATE_ENV.items():
             bargs.extend(['-e', '{0}={1}'.format(key, value)])
-        bargs.extend([get_default_execution_environment().image])
+        ee = get_default_execution_environment()
+        bargs.extend([ee.image])
+        cred = ee.credential
+        if cred.has_inputs(field_names=('host', 'username', 'password')):
+            path = functioning_dir(self.source)
+            with open(path + '/auth.json', 'w') as authfile:
+                os.chmod(authfile.name, stat.S_IRUSR | stat.S_IWUSR)
+
+                host = cred.get_input('host')
+                username = cred.get_input('username')
+                password = cred.get_input('password')
+                token = "{}:{}".format(username, password)
+                auth_data = {'auths': {host: {'auth': b64encode(token.encode('utf-8')).decode('utf-8')}}}
+                authfile.write(json.dumps(auth_data, indent=4))
+            bargs.append(f'--authfile={authfile.name}')
+        else:
+            raise RuntimeError('Please recheck that your host, username, and password fields are all filled.')
+        pull = ee.pull
+        if pull:
+            bargs.append(f'--pull={pull}')
```

Comment:
are you fearful of refactoring where-ever else we do this same thing to use a function so we can re-use it here? Is the other one mangled up with other code that its hard to pick it out?

Reply:
Sure, I think it could be refactored a bit.

Reply:
yeah if you want to refactor with the stuff I wrote we could probably consolidate nicely